package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,185 @@
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+EDR auxiliary reader --- :mod:`MDAnalysis.auxiliary.EDR`
+========================================================
+
+EDR files are binary files following the XDR protocol
+(https://datatracker.ietf.org/doc/html/rfc1014). They are written by
+GROMACS during simulations and contain the time-series energy data of the
+system.
+
+pyedr is a Python package ( https://github.com/mdanalysis/panedr ) that reads
+EDR binary files and returns them human-readable form as a dictionary of NumPy
+arrays. It is used by the EDR auxiliary reader to parse EDR files. As such, a
+dictionary with string keys and numpy array values is loaded into the
+EDRReader.
+
+The EDR auxiliary reader takes the output from pyedr and loads the energy data
+as auxiliary data into :class:`~MDAnalysis.core.universe.Universe`. Standalone
+usage is also possible, where the energy terms are extracted without
+associating them with the trajectory, for example, to allow easy plotting of
+the energy terms.
+
+
+
+"""
+import os
+from . import base
+import pyedr
+
+
+class EDRStep(base.AuxStep):
+    """ AuxStep class for .edr file format.
+
+    Extends the base AuxStep class to allow selection of time and
+    data-of-interest fields (by column index) from the full set of data read
+    each step.
+
+    Parameters
+    ----------
+    time_selector : int | None, optional
+        Index of column in .xvg file storing time, assumed to be in ps. Default
+        value is 0 (i.e. first column).
+    data_selector : list of int | None, optional
+        List of indices of columns in .xvg file containing data of interest to
+        be stored in ``data``. Default value is ``None``.
+    **kwargs
+        Other AuxStep options.
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxStep`
+    """
+
+    def __init__(self, time_selector="Time", data_selector=None, **kwargs):
+        super(EDRStep, self).__init__(time_selector=time_selector,
+                                      data_selector=data_selector,
+                                      **kwargs)
+
+    def _select_time(self, key):
+        """'Time' is one of the entries in the dict returned by pyedr.
+        The base AuxStep Class uses the time_selector 'Time' to return the
+        time value of each step."""
+        return self._select_data(key)
```

Comment:
@BFedder Generally test-driven development is the easiest way to get stuff done, so I would suggest that your next step be to concentrate on getting the tests done. Then you can add missing features / documentation / etc...